web/src/test/java/org/springframework/security/web/method/annotation/AuthenticationPrincipalArgumentResolverTests.java

```diff
@@ -199,6 +200,7 @@ public void resolveArgumentObject() throws Exception {
 	public void resolveArgumentCustomMetaAnnotation() throws Exception {
 		CustomUserPrincipal principal = new CustomUserPrincipal();
 		setAuthenticationPrincipal(principal);
+		this.resolver.setTemplateDefaults(new AnnotationTemplateExpressionDefaults());
```

Comment:
Sorry, I misunderstood your question earlier. This line should not be needed. Please see my related comment for details on how to address this.